internal/util/k8sutil/options.go

```diff
@@ -0,0 +1,39 @@
+// Copyright 2019 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package k8sutil
+
+import (
+	"k8s.io/apimachinery/pkg/fields"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// MatchingFields implements the client.ListOption and client.DeleteAllOfOption
+// interfaces so fields.Selector can be used directly in client.List and
+// client.DeleteAllOf.
+type MatchingFields struct {
+	Sel fields.Selector
+}
+
+var _ client.ListOption = MatchingFields{}
+
+func (m MatchingFields) ApplyToList(opts *client.ListOptions) {
+	opts.FieldSelector = m.Sel
+}
+
+var _ client.DeleteAllOfOption = MatchingFields{}
+
+func (m MatchingFields) ApplyToDeleteAllOf(opts *client.DeleteAllOfOptions) {
+	opts.FieldSelector = m.Sel
+}
```

Comment:
I didn't look too closely, but is this different than [`client.MatchingFields`](https://github.com/kubernetes-sigs/controller-runtime/blob/master/pkg/client/options.go#L326)?

If so should we see if we can contribute our differences upstream?

Reply:
Yes, `client.MatchingFields` is a [`fields.Set`](https://godoc.org/k8s.io/apimachinery/pkg/fields#Set), while this directly takes a parsed [`fields.Selector`](https://godoc.org/k8s.io/apimachinery/pkg/fields#Selector). I will make an upstream PR.

Reply:
Now that v0.2.0 has been released, we should keep this code here and remove it if/when it is merged into controller-runtime. I made an issue to see if we should submit a PR: https://github.com/kubernetes-sigs/controller-runtime/issues/576.

Reply:
:+1: Sounds good. Maybe make a note here about that.